src/Grammar.lhs

```diff
@@ -34,7 +34,14 @@ Here is our mid-section datatype
 
 > type Name = Int
 
-> type Production = (Name,[Name],(String,[Int]),Priority)
+> data Production
+>       = Production Name [Name] (String,[Int]) Priority
+
+#ifdef DEBUG
+
+>       deriving Show
```

Comment:
Why not always derive?

Reply:
Ahh `Name`. I thought this was all Prelude stuff stuff in there, but I was wrong. Nevermind, :+1:.